backends/tests/test_sycl_device_interface.cpp

```diff
@@ -0,0 +1,404 @@
+//===----- test_sycl_device_interface.cpp - dpctl-C_API interface -*- C++ -*-===//
+//
+//               Python Data Parallel Processing Library (PyDPPL)
+//
+// Copyright 2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file has unit test cases for functions defined in
+/// dppl_sycl_device_interface.h.
+///
+//===----------------------------------------------------------------------===//
+
+#include "dppl_sycl_device_interface.h"
+#include "dppl_sycl_queue_interface.h"
+#include "dppl_sycl_queue_manager.h"
+#include "dppl_utils.h"
+
+#include <gtest/gtest.h>
+#include <CL/sycl.hpp>
+
+using namespace cl::sycl;
+
+
+struct TestDPPLSyclDeviceInterface : public ::testing::Test
+{
+    DPPLSyclDeviceRef OpenCL_cpu = nullptr;
+    DPPLSyclDeviceRef OpenCL_gpu = nullptr;
+    DPPLSyclDeviceRef OpenCL_level_zero = nullptr;
```

Comment:
Call it Level0_gpu. OpenCl and Level0 are different driver stacks for the same GPU device.